emcc

```diff
@@ -735,6 +735,20 @@ try:
   # counter for the next index that should be used.
   next_arg_index = len(newargs)
 
+  pthreads = 'USE_PTHREADS=1' in settings_changes
+  if pthreads:
+    settings_changes.append('USE_PTHREADS=1')
```

Comment:
this line seems unnecessary, since we reach it only if that was true anyhow?

in general, I think this code should be moved to around line 996, **after** the `settings_changes` have been applied. Then just test `Settings.USE_PTHREADS`.